src/dotnet/commands/dotnet-build/DotnetProjectBuilder.cs

```diff
@@ -116,17 +116,27 @@ private void CopyCompilationOutput(OutputPaths outputPaths)
 
         private void MakeRunnable(ProjectGraphNode graphNode)
         {
-            var runtimeContext = graphNode.ProjectContext.ProjectFile.HasRuntimeOutput(_args.ConfigValue) ?
-                _args.Workspace.GetRuntimeContext(graphNode.ProjectContext, _args.GetRuntimes()) :
-                graphNode.ProjectContext;
+            foreach (var runtimeContext in CreateRuntimeContexts(graphNode))
+            {
+                var outputPaths = runtimeContext.GetOutputPaths(_args.ConfigValue, _args.BuildBasePathValue, _args.OutputValue);
+                var libraryExporter = runtimeContext.CreateExporter(_args.ConfigValue, _args.BuildBasePathValue);
+
+                CopyCompilationOutput(outputPaths);
 
-            var outputPaths = runtimeContext.GetOutputPaths(_args.ConfigValue, _args.BuildBasePathValue, _args.OutputValue);
-            var libraryExporter = runtimeContext.CreateExporter(_args.ConfigValue, _args.BuildBasePathValue);
+                var executable = new Executable(runtimeContext, outputPaths, libraryExporter, _args.ConfigValue);
+                executable.MakeCompilationOutputRunnable();
+            }
+        }
 
-            CopyCompilationOutput(outputPaths);
+        public IEnumerable<ProjectContext> CreateRuntimeContexts(ProjectGraphNode graphNode)
```

Comment:
I'd move this to the workspace context, near the other runtime context factory method